{
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.12"
  },
  "name": "",
  "signature": "sha256:7ab3de101e90efa1ee9f304b75ed6c704d1c1b2e4f711aa6e1443f065ea0caa1"
 },
 "nbformat": 3,
 "nbformat_minor": 0,
 "worksheets": [
  {
   "cells": [
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "%matplotlib inline\n",
      "import pandas as pd"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 1
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "suppT2 = pd.read_csv('/home/vpt/data/benchmark/supplementary-Table-S2-The-testing-missense-variants-of-benchmark-datasets.tsv',sep = '\\t')\n",
      "clinvar = pd.read_csv('/home/vpt/data/clinvar/clinvar_variant_summary_20190624.txt.gz',sep='\\t')"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stderr",
       "text": [
        "/usr/lib/python2.7/dist-packages/IPython/core/interactiveshell.py:2741: DtypeWarning: Columns (18) have mixed types. Specify dtype option on import or set low_memory=False.\n",
        "  interactivity=interactivity, compiler=compiler)\n"
       ]
      }
     ],
     "prompt_number": 2
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "suppT2.rename(columns = {'Data source':'Data_source'},inplace = True)\n",
      "suppT2.head()"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<div style=\"max-width:1500px;overflow:auto;\">\n",
        "<style scoped>\n",
        "    .dataframe tbody tr th:only-of-type {\n",
        "        vertical-align: middle;\n",
        "    }\n",
        "\n",
        "    .dataframe tbody tr th {\n",
        "        vertical-align: top;\n",
        "    }\n",
        "\n",
        "    .dataframe thead th {\n",
        "        text-align: right;\n",
        "    }\n",
        "</style>\n",
        "<table border=\"1\" class=\"dataframe\">\n",
        "  <thead>\n",
        "    <tr style=\"text-align: right;\">\n",
        "      <th></th>\n",
        "      <th>Data_source</th>\n",
        "      <th>Chr</th>\n",
        "      <th>Start</th>\n",
        "      <th>End</th>\n",
        "      <th>Ref</th>\n",
        "      <th>Alt</th>\n",
        "      <th>Gene_symbol</th>\n",
        "      <th>Label</th>\n",
        "    </tr>\n",
        "  </thead>\n",
        "  <tbody>\n",
        "    <tr>\n",
        "      <th>0</th>\n",
        "      <td>ClinVar</td>\n",
        "      <td>chr15</td>\n",
        "      <td>23890816</td>\n",
        "      <td>23890816</td>\n",
        "      <td>C</td>\n",
        "      <td>T</td>\n",
        "      <td>MAGEL2</td>\n",
        "      <td>Benign</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>1</th>\n",
        "      <td>ClinVar</td>\n",
        "      <td>chr2</td>\n",
        "      <td>84756049</td>\n",
        "      <td>84756049</td>\n",
        "      <td>G</td>\n",
        "      <td>A</td>\n",
        "      <td>DNAH6</td>\n",
        "      <td>Benign</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>2</th>\n",
        "      <td>ClinVar</td>\n",
        "      <td>chr2</td>\n",
        "      <td>179518937</td>\n",
        "      <td>179518937</td>\n",
        "      <td>T</td>\n",
        "      <td>C</td>\n",
        "      <td>TTN</td>\n",
        "      <td>Benign</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>3</th>\n",
        "      <td>ClinVar</td>\n",
        "      <td>chr8</td>\n",
        "      <td>10467652</td>\n",
        "      <td>10467652</td>\n",
        "      <td>G</td>\n",
        "      <td>C</td>\n",
        "      <td>RP1L1</td>\n",
        "      <td>Benign</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>4</th>\n",
        "      <td>ClinVar</td>\n",
        "      <td>chr17</td>\n",
        "      <td>76497920</td>\n",
        "      <td>76497920</td>\n",
        "      <td>C</td>\n",
        "      <td>A</td>\n",
        "      <td>DNAH17</td>\n",
        "      <td>Benign</td>\n",
        "    </tr>\n",
        "  </tbody>\n",
        "</table>\n",
        "</div>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 3,
       "text": [
        "  Data_source    Chr      Start        End Ref Alt Gene_symbol   Label\n",
        "0     ClinVar  chr15   23890816   23890816   C   T      MAGEL2  Benign\n",
        "1     ClinVar   chr2   84756049   84756049   G   A       DNAH6  Benign\n",
        "2     ClinVar   chr2  179518937  179518937   T   C         TTN  Benign\n",
        "3     ClinVar   chr8   10467652   10467652   G   C       RP1L1  Benign\n",
        "4     ClinVar  chr17   76497920   76497920   C   A      DNAH17  Benign"
       ]
      }
     ],
     "prompt_number": 3
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "# Sources:\n",
      "\n",
      "\n",
      "#### 1) *  Clinvar es clinvar ;)\n",
      "\n",
      "#### 2) * Notar que TP53 + ICGC  son ambas de Cancer. \n",
      "* Conforman un solo dataset de validacion\n",
      "\n",
      "#### 3) * PPARG gen \n",
      "* (peroxisome proliferator activated receptor \u03b3)  experimentalmente,  evaluaron todas las posibles sustituciones de aminoacidos. Este conforma un dataset de evaluacion per-se.  MITER database\n",
      "\n",
      "\n",
      "#### 4) * De novo (autismo vs control) \n",
      "* SSC 2508 pacietes con autismo vs 1911 pacientes en grupo control"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "sources = suppT2.Data_source.value_counts().index\n",
      "suppT2.Data_source.value_counts()\n"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 4,
       "text": [
        "ClinVar    4880\n",
        "SSC        2757\n",
        "PPARG      2533\n",
        "TP53       1014\n",
        "ICGC        811\n",
        "Name: Data_source, dtype: int64"
       ]
      }
     ],
     "prompt_number": 4
    },
    {
     "cell_type": "heading",
     "level": 1,
     "metadata": {},
     "source": [
      "Minimal Clinvar caracterization"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "supp_clnv = suppT2[suppT2.Data_source=='ClinVar']\n",
      "supp_clnv.Label.value_counts()"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 5,
       "text": [
        "Benign        2782\n",
        "Pathogenic    2098\n",
        "Name: Label, dtype: int64"
       ]
      }
     ],
     "prompt_number": 5
    },
    {
     "cell_type": "heading",
     "level": 1,
     "metadata": {},
     "source": [
      "Label caracterization per source"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "suppT2.groupby(['Data_source'])['Label'].value_counts()"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 32,
       "text": [
        "Data_source  Label                       \n",
        "ClinVar      Benign                          2782\n",
        "             Pathogenic                      2098\n",
        "ICGC         Likely pathogenic                429\n",
        "             Pathogenic/Likely pathogenic     189\n",
        "             Pathogenic                       145\n",
        "             Benign                            38\n",
        "             Likely benign                     10\n",
        "PPARG        not pathogenic                  2386\n",
        "             pathogenic                       147\n",
        "SSC          asd                             1650\n",
        "             control                         1107\n",
        "TP53         functional                       537\n",
        "             non-functional                   477\n",
        "Name: Label, dtype: int64"
       ]
      }
     ],
     "prompt_number": 32
    },
    {
     "cell_type": "heading",
     "level": 2,
     "metadata": {},
     "source": [
      "Este diccionario es acorde al paper   (Jinchen Li et al 2018)"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "patho_beningn = {'Benign':'Benign',\n",
      "'Pathogenic':'Pathogenic',\n",
      "'Likely pathogenic':'Pathogenic',\n",
      "'Pathogenic/Likely pathogenic':'Pathogenic',\n",
      "'Likely benign':'Benign',\n",
      "'not pathogenic':'Pathogenic',\n",
      "'pathogenic':'Pathogenic',\n",
      "'asd':'Pathogenic',\n",
      "'control':'Benign',\n",
      "'functional':'Benign',   \n",
      "'non-functional':'Pathogenic'}"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 33
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "# Perdemos a lo susmo 4474  Benignas\n",
      "*  si es que fueran todas raras y con algun homozigota en gnomad...\n",
      "\n",
      "#  Perdemos a lo sumo 7521  Patogenicas \n",
      "* uno pensaria que solo pierde las de clinvar (unas 2000) pero Clinvar pudo haberse actualizado desde entonces y pudo haber incorporado algunas de las 5500 patogenicas que vienen de otros datasets..."
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "suppT2.Label.replace(patho_beningn).value_counts()"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 34,
       "text": [
        "Pathogenic    7521\n",
        "Benign        4474\n",
        "Name: Label, dtype: int64"
       ]
      }
     ],
     "prompt_number": 34
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "# bias por gen: (pendiente ver qu\u00e9 genes componen la cola de la distribuci\u00f3n, circularidad de tipo 2)\n",
      "supp_clnv.Gene_symbol.value_counts().hist(bins = 100,log = True)#.value_counts().head(10)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 41,
       "text": [
        "<matplotlib.axes._subplots.AxesSubplot at 0x7f456ba3cb50>"
       ]
      },
      {
       "metadata": {},
       "output_type": "display_data",
       "png": "iVBORw0KGgoAAAANSUhEUgAAAXoAAAD8CAYAAAB5Pm/hAAAABHNCSVQICAgIfAhkiAAAAAlwSFlz\nAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDIuMi4yLCBo\ndHRwOi8vbWF0cGxvdGxpYi5vcmcvhp/UCwAADhdJREFUeJzt3V+IXOd5x/HvU7vOhdVumzgsQTZd\npxKmqg1OvMQJDWUFpZZsFKchNBKmTYqaxSUuLeSiMumFb0rcC7dg6iRssVFaghfj/rMtFacNHYLA\nbW0F17YqVCu2iiXciDSw7Qha1+7Tiz1SZrYa78zujGbnme8Hlp1z5syZdx7O/ph95p1zIjORJNX1\nI+MegCRptAx6SSrOoJek4gx6SSrOoJek4gx6SSrOoJek4gx6SSrOoJek4q4e9wAArrvuupybmxv4\ncRcuXODaa68d/oAmlPXoZj26WY9uFepx/Pjx72fm+9fbbksE/dzcHC+88MLAj2u1WiwsLAx/QBPK\nenSzHt2sR7cK9YiIf+1nO1s3klScQS9JxRn0klScQS9JxRn0klScQS9JxRn0klScQS9JxW2JL0xt\nxtyhI5dun3nwrjGORJK2Jt/RS1JxBr0kFWfQS1JxQw/6iPiZiPhaRDwZEb8x7P1LkgbTV9BHxGMR\ncT4iXlmzfk9EnIqI0xFxCCAzT2bmvcAvAz83/CFLkgbR7zv6w8CezhURcRXwCLAX2AUciIhdzX2f\nAI4AR4c2UknShkRm9rdhxBzwTGbe3Cx/DHggM+9olu8HyMwvdzzmSGZeds5jRCwCiwCzs7O3LS8v\nDzz4drvN6yvvXFq+ZfvMwPuopN1us23btnEPY8uwHt2sR7cK9di9e/fxzJxfb7vNzKPfDrzRsXwW\nuD0iFoBPAe/hXd7RZ+YSsAQwPz+fG7kAQKvV4qFjFy4tn7ln8H1UUuFCCsNkPbpZj27TVI+hf2Eq\nM1tAa9j7lSRtzGZm3ZwDbuhYvr5Z17eI2BcRSysrK5sYhiTp3Wwm6J8HdkbEjRFxDbAfeGqQHWTm\n05m5ODMz3b11SRqlfqdXPg48B9wUEWcj4mBmvg3cBzwLnASeyMwToxuqJGkj+urRZ+aBHuuP4hRK\nSdrSxnoKBHv0kjR6Yw16e/SSNHqe1EySirN1I0nF2bqRpOJs3UhScQa9JBVnj16SirNHL0nF2bqR\npOIMekkqzqCXpOIMekkqzlk3klScs24kqThbN5JUnEEvScUZ9JJUnEEvScU560aSinPWjSQVZ+tG\nkooz6CWpOINekooz6CWpOINekooz6CWpOOfRS1JxzqOXpOJs3UhScQa9JBVn0EtScQa9JBVn0EtS\ncQa9JBVn0EtScQa9JBVn0EtScZ4CQZKK8xQIklScrRtJKs6gl6Tirh73AIZp7tCRS7fPPHjXGEci\nSVuH7+glqTiDXpKKM+glqTiDXpKKM+glqTiDXpKKM+glqTiDXpKKM+glqTiDXpKKG8kpECLik8Bd\nwI8Dj2bmN0fxPJKk9fX9jj4iHouI8xHxypr1eyLiVEScjohDAJn5l5n5eeBe4DPDHbIkaRCDtG4O\nA3s6V0TEVcAjwF5gF3AgInZ1bPK7zf2SpDHpO+gz89vAD9as/ghwOjNfy8y3gGXg7lj1+8BfZ+Z3\nhjdcSdKgNtuj3w680bF8Frgd+E3gF4CZiNiRmV9b+8CIWAQWAWZnZ2m1WgM/ebvd5ou3vHPZ+zay\nv0nXbren8nX3Yj26WY9u01SPkXwYm5kPAw+vs80SsAQwPz+fCwsLAz9Pq9XioWMXLnvfmXsG39+k\na7VabKSOVVmPbtaj2zTVY7PTK88BN3QsX9+skyRtEZsN+ueBnRFxY0RcA+wHnur3wRGxLyKWVlZW\nNjkMSVIvg0yvfBx4DrgpIs5GxMHMfBu4D3gWOAk8kZkn+t1nZj6dmYszMzODjluS1Ke+e/SZeaDH\n+qPA0aGNSJI0VGM9BYKtG0kavbEGva0bSRo9T2omScWNZB79VjB36Mil22cevGuMI5Gk8bJHL0nF\n2aOXpOLs0UtScQa9JBVnj16SirNHL0nF2bqRpOIMekkqzqCXpOL8MFaSivPDWEkqztaNJBVn0EtS\ncQa9JBVn0EtScc66kaTinHUjScXZupGk4gx6SSqu7DVje/FaspKmzVQEfWe4S9K0sXUjScUZ9JJU\nnPPoJak459FLUnG2biSpOINekooz6CWpOINekooz6CWpOINekooz6CWpOINekooz6CWpOE+BIEnF\neQoESSrO1o0kFWfQS1JxBr0kFWfQS1JxBr0kFWfQS1JxBr0kFWfQS1JxBr0kFWfQS1JxBr0kFWfQ\nS1JxBr0kFWfQS1JxQw/6iPhgRDwaEU8Oe9+SpMH1FfQR8VhEnI+IV9as3xMRpyLidEQcAsjM1zLz\n4CgGO2pzh45c+pGkKvp9R38Y2NO5IiKuAh4B9gK7gAMRsWuoo5MkbVpkZn8bRswBz2Tmzc3yx4AH\nMvOOZvl+gMz8crP8ZGZ++l32twgsAszOzt62vLw88ODb7Tavr7wz8OMuumV795WtXj630vO+SdBu\nt9m2bdu4h7FlWI9u1qNbhXrs3r37eGbOr7fd1Zt4ju3AGx3LZ4HbI+J9wO8BH4qI+y8G/1qZuQQs\nAczPz+fCwsLAA2i1Wjx07MLAj7vozD3dz/m5jpbN2vsmQavVYiN1rMp6dLMe3aapHpsJ+svKzH8H\n7h32fiVJG7OZoD8H3NCxfH2zrm8RsQ/Yt2PHjk0MY+P80FXSNNjM9MrngZ0RcWNEXAPsB54aZAeZ\n+XRmLs7MTF4/XJImRb/TKx8HngNuioizEXEwM98G7gOeBU4CT2TmidENVZK0EX21bjLzQI/1R4Gj\nQx2RJGmoxnoKhIjYFxFLKysr628sSdqQsQa9PXpJGj1PaiZJxdm6kaTibN1IUnG2biSpOINekoqz\nRy9Jxdmjl6TibN1IUnEGvSQVZ9BLUnEGvSQV56wbSSrOWTeSVJytG0kqzqCXpOIMekkqzqCXpOL6\numbsqETEPmDfjh07xjmMdc0dOnLp9pkH7xrjSCRpcM66kaTibN1IUnEGvSQVZ9BLUnEGvSQVZ9BL\nUnEGvSQV5zz6HjrnzkvSJHMevSQVZ+tGkooz6CWpOINekooz6CWpOINekooz6CWpOINekooz6CWp\nOINekorzFAgD6nVqhM5LDPazjSRdKZ4CQZKKs3UjScUZ9JJUnEEvScUZ9JJUnEEvScUZ9JJUnEEv\nScUZ9JJUnEEvScUZ9JJUnEEvScUZ9JJUnEEvScUN/TTFEXEt8BXgLaCVmd8Y9nNIkvrX1zv6iHgs\nIs5HxCtr1u+JiFMRcToiDjWrPwU8mZmfBz4x5PFKkgbUb+vmMLCnc0VEXAU8AuwFdgEHImIXcD3w\nRrPZO8MZpiRpo/oK+sz8NvCDNas/ApzOzNcy8y1gGbgbOMtq2Pe9f0nS6ERm9rdhxBzwTGbe3Cx/\nGtiTmb/eLP8KcDvwO8AfAf8FHOvVo4+IRWARYHZ29rbl5eWBB99ut3l9ZWv803DL9h9eJevlcyvr\nbtOpc/t+tum1z3a7zbZt2/reZ3Wd9ZD1WGuc9RjW3+fu3buPZ+b8etsN/cPYzLwA/Fof2y0BSwDz\n8/O5sLAw8HO1Wi0eOnZh4MeNwpl7Fi7d/lyva8Z2bNOpc/t+tum1z1arxcU69rPP6jrrIeux1jjr\ncaX/PjfTWjkH3NCxfH2zTpK0hWwm6J8HdkbEjRFxDbAfeGqQHUTEvohYWlm5fFtCkrR5/U6vfBx4\nDrgpIs5GxMHMfBu4D3gWOAk8kZknBnnyzHw6MxdnZqazhyxJV0JfPfrMPNBj/VHg6FBHJEkaqrFO\nf7R1I0mjN9agt3UjSaPnF5okqTiDXpKK6/ubsSN58oh9wD7gM8CrG9jFdcD3hzqoyWY9ulmPbtaj\nW4V6/FRmvn+9jcYa9JsVES/08/XfaWE9ulmPbtaj2zTVw9aNJBVn0EtScZMe9EvjHsAWYz26WY9u\n1qPb1NRjonv0kqT1Tfo7eknSOiYy6Htcq3aqRMSZiHg5Il6MiBeade+NiL+JiFeb3z857nGOyuWu\nY9zr9ceqh5vj5aWI+PD4Rj4aPerxQESca46RFyPizo777m/qcSoi7hjPqEcnIm6IiL+LiH+OiBMR\n8VvN+qk8RiYu6N/lWrXTaHdm3toxRewQ8K3M3Al8q1mu6jBrrmNM79e/F9jZ/CwCX71CY7ySDvP/\n6wHwh80xcmtzEkKav5f9wM82j/lK83dVydvAFzNzF/BR4AvN657KY2Tigp7e16rVah2+3tz+OvDJ\nMY5lpHpcx7jX678b+JNc9ffAT0TEB67MSK+MHvXo5W5gOTP/OzNfB06z+ndVRma+mZnfaW7/J6un\nUt/OlB4jkxj024E3OpbPNuumTQLfjIjjzfV3AWYz883m9r8Bs+MZ2tj0ev3TfMzc17QiHuto5U1V\nPZrrXX8I+Aem9BiZxKDXqo9n5odZ/ZfzCxHx85135up0qqmdUjXtr7/xVeCngVuBN4GHxjucKy8i\ntgF/Bvx2Zv5H533TdIxMYtB7rVogM881v88Df8Hqv97fu/jvZvP7/PhGOBa9Xv9UHjOZ+b3MfCcz\n/xf4Y37YnpmKekTEj7Ia8t/IzD9vVk/lMTKJQb/pa9VOuoi4NiJ+7OJt4BeBV1itw2ebzT4L/NV4\nRjg2vV7/U8CvNjMrPgqsdPz7XtaaHvMvsXqMwGo99kfEeyLiRlY/gPzHKz2+UYqIAB4FTmbmH3Tc\nNZ3HSGZO3A9wJ/AvwHeBL417PGN4/R8E/qn5OXGxBsD7WJ1J8Crwt8B7xz3WEdbgcVbbEf/Daj/1\nYK/XDwSrM7W+C7wMzI97/FeoHn/avN6XWA2yD3Rs/6WmHqeAveMe/wjq8XFW2zIvAS82P3dO6zHi\nN2MlqbhJbN1IkgZg0EtScQa9JBVn0EtScQa9JBVn0EtScQa9JBVn0EtScf8Hf0CY9uNFcEsAAAAA\nSUVORK5CYII=\n",
       "text": [
        "<Figure size 432x288 with 1 Axes>"
       ]
      }
     ],
     "prompt_number": 41
    },
    {
     "cell_type": "heading",
     "level": 1,
     "metadata": {},
     "source": [
      "No hay algun indel filtrado..."
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "any(suppT2.Start != suppT2.End)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 28,
       "text": [
        "False"
       ]
      }
     ],
     "prompt_number": 28
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "# Clinvar: count number of review cases"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "supp_clnv.columns"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 10,
       "text": [
        "Index([u'Data_source', u'Chr', u'Start', u'End', u'Ref', u'Alt',\n",
        "       u'Gene_symbol', u'Label'],\n",
        "      dtype='object')"
       ]
      }
     ],
     "prompt_number": 10
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "clinvar.columns"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 7,
       "text": [
        "Index([u'#AlleleID', u'Type', u'Name', u'GeneID', u'GeneSymbol', u'HGNC_ID',\n",
        "       u'ClinicalSignificance', u'ClinSigSimple', u'LastEvaluated',\n",
        "       u'RS# (dbSNP)', u'nsv/esv (dbVar)', u'RCVaccession', u'PhenotypeIDS',\n",
        "       u'PhenotypeList', u'Origin', u'OriginSimple', u'Assembly',\n",
        "       u'ChromosomeAccession', u'Chromosome', u'Start', u'Stop',\n",
        "       u'ReferenceAllele', u'AlternateAllele', u'Cytogenetic', u'ReviewStatus',\n",
        "       u'NumberSubmitters', u'Guidelines', u'TestedInGTR', u'OtherIDs',\n",
        "       u'SubmitterCategories', u'VariationID'],\n",
        "      dtype='object')"
       ]
      }
     ],
     "prompt_number": 7
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "supp_clnv['ind'] = supp_clnv.Chr.map(str) + '-' + supp_clnv.Start.map(str) + '-' + supp_clnv.Ref.map(str) + '-' + supp_clnv.Alt.map(str)\n",
      "clinvar['ind'] = 'chr' + clinvar.Chromosome.map(str) + '-' + clinvar.Start.map(str) + '-' + clinvar.ReferenceAllele.map(str) + '-' + clinvar.AlternateAllele.map(str)\n",
      "supp_clnv.set_index(['ind'],inplace = True)\n",
      "clinvar.set_index(['ind'],inplace = True)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 19
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "suppT2_clinvar_updated = pd.merge(supp_clnv,clinvar[clinvar.Assembly == 'GRCh37'][[u'ClinicalSignificance', u'ClinSigSimple','LastEvaluated',u'Assembly',\\\n",
      "                                                     u'ReferenceAllele', u'AlternateAllele', u'Cytogenetic', u'ReviewStatus',u'NumberSubmitters',\n",
      "                                                     u'Guidelines', u'TestedInGTR', u'OtherIDs',u'SubmitterCategories', u'VariationID']],\n",
      "                                  left_index=True,right_index = True,how = 'left')\n"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 31
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "suppT2_clinvar_updated.groupby(['Label'])['ClinSigSimple'].value_counts().unstack()"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<div style=\"max-width:1500px;overflow:auto;\">\n",
        "<style scoped>\n",
        "    .dataframe tbody tr th:only-of-type {\n",
        "        vertical-align: middle;\n",
        "    }\n",
        "\n",
        "    .dataframe tbody tr th {\n",
        "        vertical-align: top;\n",
        "    }\n",
        "\n",
        "    .dataframe thead th {\n",
        "        text-align: right;\n",
        "    }\n",
        "</style>\n",
        "<table border=\"1\" class=\"dataframe\">\n",
        "  <thead>\n",
        "    <tr style=\"text-align: right;\">\n",
        "      <th>ClinSigSimple</th>\n",
        "      <th>-1.0</th>\n",
        "      <th>0.0</th>\n",
        "      <th>1.0</th>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Label</th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "    </tr>\n",
        "  </thead>\n",
        "  <tbody>\n",
        "    <tr>\n",
        "      <th>Benign</th>\n",
        "      <td>NaN</td>\n",
        "      <td>2759.0</td>\n",
        "      <td>19.0</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>Pathogenic</th>\n",
        "      <td>11.0</td>\n",
        "      <td>4.0</td>\n",
        "      <td>2081.0</td>\n",
        "    </tr>\n",
        "  </tbody>\n",
        "</table>\n",
        "</div>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 34,
       "text": [
        "ClinSigSimple  -1.0     0.0     1.0\n",
        "Label                              \n",
        "Benign          NaN  2759.0    19.0\n",
        "Pathogenic     11.0     4.0  2081.0"
       ]
      }
     ],
     "prompt_number": 34
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "print supp_clnv.shape\n",
      "print clinvar.Assembly.value_counts()"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "(4880, 8)\n",
        "GRCh37    501313\n",
        "GRCh38    472128\n",
        "NCBI36     19625\n",
        "Name: Assembly, dtype: int64"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n"
       ]
      }
     ],
     "prompt_number": 37
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "suppT2_clinvar_updated.Ref"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<div style=\"max-width:1500px;overflow:auto;\">\n",
        "<style scoped>\n",
        "    .dataframe tbody tr th:only-of-type {\n",
        "        vertical-align: middle;\n",
        "    }\n",
        "\n",
        "    .dataframe tbody tr th {\n",
        "        vertical-align: top;\n",
        "    }\n",
        "\n",
        "    .dataframe thead th {\n",
        "        text-align: right;\n",
        "    }\n",
        "</style>\n",
        "<table border=\"1\" class=\"dataframe\">\n",
        "  <thead>\n",
        "    <tr style=\"text-align: right;\">\n",
        "      <th></th>\n",
        "      <th>Data_source</th>\n",
        "      <th>Chr</th>\n",
        "      <th>Start_x</th>\n",
        "      <th>End</th>\n",
        "      <th>Ref</th>\n",
        "      <th>Alt</th>\n",
        "      <th>Gene_symbol</th>\n",
        "      <th>Label</th>\n",
        "      <th>Chromosome</th>\n",
        "      <th>Start_y</th>\n",
        "      <th>...</th>\n",
        "      <th>ReferenceAllele</th>\n",
        "      <th>AlternateAllele</th>\n",
        "      <th>Cytogenetic</th>\n",
        "      <th>ReviewStatus</th>\n",
        "      <th>NumberSubmitters</th>\n",
        "      <th>Guidelines</th>\n",
        "      <th>TestedInGTR</th>\n",
        "      <th>OtherIDs</th>\n",
        "      <th>SubmitterCategories</th>\n",
        "      <th>VariationID</th>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>ind</th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "      <th></th>\n",
        "    </tr>\n",
        "  </thead>\n",
        "  <tbody>\n",
        "    <tr>\n",
        "      <th>chr1-100340782-G-T</th>\n",
        "      <td>ClinVar</td>\n",
        "      <td>chr1</td>\n",
        "      <td>100340782</td>\n",
        "      <td>100340782</td>\n",
        "      <td>G</td>\n",
        "      <td>T</td>\n",
        "      <td>AGL</td>\n",
        "      <td>Benign</td>\n",
        "      <td>NaN</td>\n",
        "      <td>NaN</td>\n",
        "      <td>...</td>\n",
        "      <td>NaN</td>\n",
        "      <td>NaN</td>\n",
        "      <td>NaN</td>\n",
        "      <td>NaN</td>\n",
        "      <td>NaN</td>\n",
        "      <td>NaN</td>\n",
        "      <td>NaN</td>\n",
        "      <td>NaN</td>\n",
        "      <td>NaN</td>\n",
        "      <td>NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>chr1-100358107-A-G</th>\n",
        "      <td>ClinVar</td>\n",
        "      <td>chr1</td>\n",
        "      <td>100358107</td>\n",
        "      <td>100358107</td>\n",
        "      <td>A</td>\n",
        "      <td>G</td>\n",
        "      <td>AGL</td>\n",
        "      <td>Benign</td>\n",
        "      <td>NaN</td>\n",
        "      <td>NaN</td>\n",
        "      <td>...</td>\n",
        "      <td>NaN</td>\n",
        "      <td>NaN</td>\n",
        "      <td>NaN</td>\n",
        "      <td>NaN</td>\n",
        "      <td>NaN</td>\n",
        "      <td>NaN</td>\n",
        "      <td>NaN</td>\n",
        "      <td>NaN</td>\n",
        "      <td>NaN</td>\n",
        "      <td>NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>chr1-100591478-C-T</th>\n",
        "      <td>ClinVar</td>\n",
        "      <td>chr1</td>\n",
        "      <td>100591478</td>\n",
        "      <td>100591478</td>\n",
        "      <td>C</td>\n",
        "      <td>T</td>\n",
        "      <td>SASS6</td>\n",
        "      <td>Benign</td>\n",
        "      <td>NaN</td>\n",
        "      <td>NaN</td>\n",
        "      <td>...</td>\n",
        "      <td>NaN</td>\n",
        "      <td>NaN</td>\n",
        "      <td>NaN</td>\n",
        "      <td>NaN</td>\n",
        "      <td>NaN</td>\n",
        "      <td>NaN</td>\n",
        "      <td>NaN</td>\n",
        "      <td>NaN</td>\n",
        "      <td>NaN</td>\n",
        "      <td>NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>chr1-100680410-C-T</th>\n",
        "      <td>ClinVar</td>\n",
        "      <td>chr1</td>\n",
        "      <td>100680410</td>\n",
        "      <td>100680410</td>\n",
        "      <td>C</td>\n",
        "      <td>T</td>\n",
        "      <td>DBT</td>\n",
        "      <td>Pathogenic</td>\n",
        "      <td>NaN</td>\n",
        "      <td>NaN</td>\n",
        "      <td>...</td>\n",
        "      <td>NaN</td>\n",
        "      <td>NaN</td>\n",
        "      <td>NaN</td>\n",
        "      <td>NaN</td>\n",
        "      <td>NaN</td>\n",
        "      <td>NaN</td>\n",
        "      <td>NaN</td>\n",
        "      <td>NaN</td>\n",
        "      <td>NaN</td>\n",
        "      <td>NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>chr1-100681587-A-G</th>\n",
        "      <td>ClinVar</td>\n",
        "      <td>chr1</td>\n",
        "      <td>100681587</td>\n",
        "      <td>100681587</td>\n",
        "      <td>A</td>\n",
        "      <td>G</td>\n",
        "      <td>DBT</td>\n",
        "      <td>Benign</td>\n",
        "      <td>NaN</td>\n",
        "      <td>NaN</td>\n",
        "      <td>...</td>\n",
        "      <td>NaN</td>\n",
        "      <td>NaN</td>\n",
        "      <td>NaN</td>\n",
        "      <td>NaN</td>\n",
        "      <td>NaN</td>\n",
        "      <td>NaN</td>\n",
        "      <td>NaN</td>\n",
        "      <td>NaN</td>\n",
        "      <td>NaN</td>\n",
        "      <td>NaN</td>\n",
        "    </tr>\n",
        "  </tbody>\n",
        "</table>\n",
        "<p>5 rows \u00d7 23 columns</p>\n",
        "</div>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 12,
       "text": [
        "                   Data_source   Chr    Start_x        End Ref Alt  \\\n",
        "ind                                                                  \n",
        "chr1-100340782-G-T     ClinVar  chr1  100340782  100340782   G   T   \n",
        "chr1-100358107-A-G     ClinVar  chr1  100358107  100358107   A   G   \n",
        "chr1-100591478-C-T     ClinVar  chr1  100591478  100591478   C   T   \n",
        "chr1-100680410-C-T     ClinVar  chr1  100680410  100680410   C   T   \n",
        "chr1-100681587-A-G     ClinVar  chr1  100681587  100681587   A   G   \n",
        "\n",
        "                   Gene_symbol       Label Chromosome  Start_y     ...      \\\n",
        "ind                                                                ...       \n",
        "chr1-100340782-G-T         AGL      Benign        NaN      NaN     ...       \n",
        "chr1-100358107-A-G         AGL      Benign        NaN      NaN     ...       \n",
        "chr1-100591478-C-T       SASS6      Benign        NaN      NaN     ...       \n",
        "chr1-100680410-C-T         DBT  Pathogenic        NaN      NaN     ...       \n",
        "chr1-100681587-A-G         DBT      Benign        NaN      NaN     ...       \n",
        "\n",
        "                   ReferenceAllele  AlternateAllele Cytogenetic ReviewStatus  \\\n",
        "ind                                                                            \n",
        "chr1-100340782-G-T             NaN              NaN         NaN          NaN   \n",
        "chr1-100358107-A-G             NaN              NaN         NaN          NaN   \n",
        "chr1-100591478-C-T             NaN              NaN         NaN          NaN   \n",
        "chr1-100680410-C-T             NaN              NaN         NaN          NaN   \n",
        "chr1-100681587-A-G             NaN              NaN         NaN          NaN   \n",
        "\n",
        "                   NumberSubmitters Guidelines TestedInGTR  OtherIDs  \\\n",
        "ind                                                                    \n",
        "chr1-100340782-G-T              NaN        NaN         NaN       NaN   \n",
        "chr1-100358107-A-G              NaN        NaN         NaN       NaN   \n",
        "chr1-100591478-C-T              NaN        NaN         NaN       NaN   \n",
        "chr1-100680410-C-T              NaN        NaN         NaN       NaN   \n",
        "chr1-100681587-A-G              NaN        NaN         NaN       NaN   \n",
        "\n",
        "                   SubmitterCategories VariationID  \n",
        "ind                                                 \n",
        "chr1-100340782-G-T                 NaN         NaN  \n",
        "chr1-100358107-A-G                 NaN         NaN  \n",
        "chr1-100591478-C-T                 NaN         NaN  \n",
        "chr1-100680410-C-T                 NaN         NaN  \n",
        "chr1-100681587-A-G                 NaN         NaN  \n",
        "\n",
        "[5 rows x 23 columns]"
       ]
      }
     ],
     "prompt_number": 12
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [],
     "language": "python",
     "metadata": {},
     "outputs": []
    }
   ],
   "metadata": {}
  }
 ]
}